sapl/comissoes/views.py

```diff
@@ -249,6 +280,98 @@ def get_initial(self):
+            RemovePautaView, self
+        ).get_context_data(**kwargs)
+
+        # Remove = 0; Adiciona = 1
+        context['opcao'] = 0
+
+        context['object'] = Reuniao.objects.get(pk=self.kwargs['pk'])
+        context['root_pk'] = context['object'].comissao.pk
+
+        materias_pauta = PautaReuniao.objects.filter(reuniao=context['object'])
+        materias_pk = [materia_pauta.materia.pk for materia_pauta in materias_pauta]
+        
+        context['materias'] = MateriaLegislativa.objects.filter(
+            pk__in=materias_pk
+        ).order_by('tipo', '-ano', '-numero') 
+        context['num_materias'] = len(context['materias'])
+
+        return context
+
+    def post(self, request, *args, **kwargs):
+        success_url = reverse('sapl.comissoes:reuniao_detail', kwargs={'pk':kwargs['pk']})
+        marcadas = request.POST.getlist('materia_id')
+
+        if len(marcadas) == 0:
+            msg=_('Nenhuma matéria foi selecionada.')
+            messages.add_message(request, messages.WARNING, msg)
+            return HttpResponseRedirect(success_url)
+
+        reuniao = Reuniao.objects.get(pk=kwargs['pk'])
+        for materia in MateriaLegislativa.objects.filter(id__in=marcadas):
+            PautaReuniao.objects.get(reuniao=reuniao,materia=materia).delete()
+
+        msg=_('Matéria(s) removida(s) com sucesso!')
+        messages.add_message(request, messages.SUCCESS, msg)
+        return HttpResponseRedirect(success_url)
+
+
+class AdicionaPautaView(PermissionRequiredMixin, CreateView):
+    model = PautaReuniao
+    form_class = PautaReuniaoForm
+    template_name = 'comissoes/pauta.html'
+    permission_required = ('comissoes.add_reuniao', )
+
+    def get_context_data(self, **kwargs):
+        context = super(
+            AdicionaPautaView, self
+        ).get_context_data(**kwargs)
+
+        # Adiciona = 1; Remove = 0
+        context['opcao'] = 1
+
+        context['object'] = Reuniao.objects.get(pk=self.kwargs['pk'])
+        context['root_pk'] = context['object'].comissao.pk
+
+        materias_comissao = lista_materias_comissao(context['object'].comissao.pk)
+        materias_pauta = PautaReuniao.objects.filter(reuniao=context['object'])
+
+        nao_listar = [mp.materia.pk for mp in materias_pauta]
+        context['materias'] = materias_comissao.exclude(pk__in=nao_listar)
+        context['num_materias'] = len(context['materias'])
+
+        return context
+    
+    def post(self, request, *args, **kwargs):
+        success_url = reverse('sapl.comissoes:reuniao_detail', kwargs={'pk':kwargs['pk']}) 
+        marcadas = request.POST.getlist('materia_id')
+
+        if len(marcadas) == 0:
+            msg = _('Nenhuma máteria foi selecionada.')
+            messages.add_message(request, messages.WARNING, msg)
+            return HttpResponseRedirect(success_url)
+            
+        reuniao = Reuniao.objects.get(pk=kwargs['pk'])
+        for materia in MateriaLegislativa.objects.filter(id__in=marcadas):
```

Comment:
Eu acredito que podemos usar um `bulk_create` aqui. Assim:

```suggestion
        pautas = []
        for materia in MateriaLegislativa.objects.filter(id__in=marcadas):
                 pauta = PautaReuniao()
                 pauta.reuniao = reuniao
                 pauta.materia = materia
                 pautas.append(pauta)
         PautaReuniao.objects.bulk_create(pautas)
```

A vantagem desse método é que se forem inseridas 10 pautas, ao invés de 10 comandos de INSERT no BD, será realizado somente um comando com 10 entradas. Mais eficiente e consistente (ou tudo é inserido ou nada é inserido).